### DISCUSSION TITLE
list index out of range error in DCPF model

### DISCUSSION THREAD

**Discussion by maryammoh68** [bukhsh/oats#129] [Q&A]
Hi,
I am using the DC power flow model, I have written the data from IEEE39 test case with the OATS code to a datfile but I am getting the following error:
"list index out of range"
Since I need to alter the code I have used the models I have used a modified printdata.py to write the dat file which has created the following file, but it seems one of the parameters or sets are out of bound. Can anybody help me with this since I am beginner in python and pyomo?
set B:=
1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
;
set G:=
G1
G2
G3
G4
G5
G6
G7
G8
G9
G10
;
set D:=
D1
D2
D3
D4
D5
D6
D7
D8
D9
D10
D11
D12
D13
D14
D15
D16
D17
D18
D19
D20
D21
D22
D23
D24
D25
D26
D27
D28
D29
D30
D31
D32
D33
D34
D35
D36
D37
D38
D39
;
param PD:=
D1 0.976
D2 0.0
D3 3.22
D4 5.0
D5 0.0
D6 0.0
D7 2.338
D8 5.22
D9 0.065
D10 0.0
D11 0.0
D12 0.08529999999999999
D13 0.0
D14 0.0
D15 3.2
D16 3.29
D17 0.0
D18 1.58
D19 0.0
D20 6.8
D21 2.74
D22 0.0
D23 2.475
D24 3.0860000000000003
D25 2.24
D26 1.39
D27 2.81
D28 2.06
D29 2.835
D30 0.0
D31 0.092
D32 0.0
D33 0.0
D34 0.0
D35 0.0
D36 0.0
D37 0.0
D38 0.0
D39 11.04
;
param VOLL:=
D1 100000.0
D2 100000.0
D3 100000.0
D4 100000.0
D5 100000.0
D6 100000.0
D7 100000.0
D8 100000.0
D9 100000.0
D10 100000.0
D11 100000.0
D12 100000.0
D13 100000.0
D14 100000.0
D15 100000.0
D16 100000.0
D17 100000.0
D18 100000.0
D19 100000.0
D20 100000.0
D21 100000.0
D22 100000.0
D23 100000.0
D24 100000.0
D25 100000.0
D26 100000.0
D27 100000.0
D28 100000.0
D29 100000.0
D30 100000.0
D31 100000.0
D32 100000.0
D33 100000.0
D34 100000.0
D35 100000.0
D36 100000.0
D37 100000.0
D38 100000.0
D39 100000.0
;
param baseMVA:=
100
;
set LE:=
 1 
 2;
set L:=
L1-12
L2-139
L3-23
L4-225
L5-34
L6-318
L7-45
L8-414
L9-56
L10-58
L11-67
L12-611
L13-78
L14-89
L15-939
L16-1011
L17-1013
L18-1314
L19-1415
L20-1516
L21-1617
L22-1619
L23-1621
L24-1624
L25-1718
L26-1727
L27-2122
L28-2223
L29-2324
L30-2526
L31-2627
L32-2628
L33-2629
L34-2829
;
set TRANSF:= 
T1-230
T2-631
T3-1032
T4-1211
T5-1213
T6-1920
T7-1933
T8-2034
T9-2235
T10-2336
T11-2537
T12-2938
;
set Gbs:=
30 G1
31 G2
32 G3
33 G4
34 G5
35 G6
36 G7
37 G8
38 G9
39 G10
;
set Dbs:=
1 D1
2 D2
3 D3
4 D4
5 D5
6 D6
7 D7
8 D8
9 D9
10 D10
11 D11
12 D12
13 D13
14 D14
15 D15
16 D16
17 D17
18 D18
19 D19
20 D20
21 D21
22 D22
23 D23
24 D24
25 D25
26 D26
27 D27
28 D28
29 D29
30 D30
31 D31
32 D32
33 D33
34 D34
35 D35
36 D36
37 D37
38 D38
39 D39
;
set b0:=
31
;
param A:=
L1-12 1 1
L2-139 1 1
L3-23 1 2
L4-225 1 2
L5-34 1 3
L6-318 1 3
L7-45 1 4
L8-414 1 4
L9-56 1 5
L10-58 1 5
L11-67 1 6
L12-611 1 6
L13-78 1 7
L14-89 1 8
L15-939 1 9
L16-1011 1 10
L17-1013 1 10
L18-1314 1 13
L19-1415 1 14
L20-1516 1 15
L21-1617 1 16
L22-1619 1 16
L23-1621 1 16
L24-1624 1 16
L25-1718 1 17
L26-1727 1 17
L27-2122 1 21
L28-2223 1 22
L29-2324 1 23
L30-2526 1 25
L31-2627 1 26
L32-2628 1 26
L33-2629 1 26
L34-2829 1 28
L1-12 2 2
L2-139 2 39
L3-23 2 3
L4-225 2 25
L5-34 2 4
L6-318 2 18
L7-45 2 5
L8-414 2 14
L9-56 2 6
L10-58 2 8
L11-67 2 7
L12-611 2 11
L13-78 2 8
L14-89 2 9
L15-939 2 39
L16-1011 2 11
L17-1013 2 13
L18-1314 2 14
L19-1415 2 15
L20-1516 2 16
L21-1617 2 17
L22-1619 2 19
L23-1621 2 21
L24-1624 2 24
L25-1718 2 18
L26-1727 2 27
L27-2122 2 22
L28-2223 2 23
L29-2324 2 24
L30-2526 2 26
L31-2627 2 27
L32-2628 2 28
L33-2629 2 29
L34-2829 2 29
;
param PG:=
G1 2.5
G2 6.77871
G3 6.5
G4 6.32
G5 5.08
G6 6.5
G7 5.6
G8 5.4
G9 8.3
G10 10.0
;
param BL:=
L1-12 -24.330900243309003
L2-139 -40.0
L3-23 -66.2251655629139
L4-225 -116.27906976744185
L5-34 -46.948356807511736
L6-318 -75.18796992481204
L7-45 -78.125
L8-414 -77.51937984496124
L9-56 -384.61538461538464
L10-58 -89.28571428571429
L11-67 -108.69565217391305
L12-611 -121.95121951219511
L13-78 -217.3913043478261
L14-89 -27.548209366391184
L15-939 -40.0
L16-1011 -232.5581395348837
L17-1013 -232.5581395348837
L18-1314 -99.00990099009901
L19-1415 -46.08294930875576
L20-1516 -106.38297872340425
L21-1617 -112.35955056179776
L22-1619 -51.282051282051285
L23-1621 -74.07407407407408
L24-1624 -169.49152542372883
L25-1718 -121.95121951219511
L26-1727 -57.80346820809249
L27-2122 -71.42857142857143
L28-2223 -104.16666666666667
L29-2324 -28.57142857142857
L30-2526 -30.959752321981423
L31-2627 -68.02721088435375
L32-2628 -21.09704641350211
L33-2629 -16.0
L34-2829 -66.2251655629139
;
param BLT:=
T1-230 -55.24861878453038
T2-631 -40.0
T3-1032 -50.0
T4-1211 -22.98850574712644
T5-1213 -22.98850574712644
T6-1920 -72.46376811594203
T7-1933 -70.4225352112676
T8-2034 -55.55555555555556
T9-2235 -69.93006993006993
T10-2336 -36.76470588235294
T11-2537 -43.10344827586207
T12-2938 -64.1025641025641
;
param PGmin:=
G1 0.0
G2 0.0
G3 0.0
G4 0.0
G5 0.0
G6 0.0
G7 0.0
G8 0.0
G9 0.0
G10 0.0
;
param PGmax:=
G1 10.4
G2 6.46
G3 7.25
G4 6.52
G5 5.08
G6 6.87
G7 5.8
G8 5.64
G9 8.65
G10 11.0
;

param SLmax:=
L1-12 6.0
L2-139 10.0
L3-23 5.0
L4-225 5.0
L5-34 5.0
L6-318 5.0
L7-45 6.0
L8-414 5.0
L9-56 12.0
L10-58 9.0
L11-67 9.0
L12-611 4.8
L13-78 9.0
L14-89 9.0
L15-939 9.0
L16-1011 6.0
L17-1013 6.0
L18-1314 6.0
L19-1415 6.0
L20-1516 6.0
L21-1617 6.0
L22-1619 6.0
L23-1621 6.0
L24-1624 6.0
L25-1718 6.0
L26-1727 6.0
L27-2122 9.0
L28-2223 6.0
L29-2324 6.0
L30-2526 6.0
L31-2627 6.0
L32-2628 6.0
L33-2629 6.0
L34-2829 6.0
;
param SLmaxT:=
T1-230 9.0
T2-631 18.0
T3-1032 9.0
T4-1211 5.0
T5-1213 5.0
T6-1920 9.0
T7-1933 9.0
T8-2034 9.0
T9-2235 9.0
T10-2336 9.0
T11-2537 9.0
T12-2938 12.0
;
param c2:=
G1 0.01
G2 0.01
G3 0.01
G4 0.01
G5 0.01
G6 0.01
G7 0.01
G8 0.01
G9 0.01
G10 0.01
;
param c1:=
G1 0.3
G2 0.3
G3 0.3
G4 0.3
G5 0.3
G6 0.3
G7 0.3
G8 0.3
G9 0.3
G10 0.3
;
param c0:=
G1 0.2
G2 0.2
G3 0.2
G4 0.2
G5 0.2
G6 0.2
G7 0.2
G8 0.2
G9 0.2
G10 0.2;

param AT:= 
T1-230 1 2
T2-631 1 6
T3-1032 1 10
T4-1211 1 12
T5-1213 1 12
T6-1920 1 19
T7-1933 1 19
T8-2034 1 20
T9-2235 1 22
T10-2336 1 23
T11-2537 1 25
T12-2938 1 29
T1-230 2 30
T2-631 2 31
T3-1032 2 32
T4-1211 2 11
T5-1213 2 13
T6-1920 2 20
T7-1933 2 33
T8-2034 2 34
T9-2235 2 35
T10-2336 2 36
T11-2537 2 37
T12-2938 2 38
;

set SHUNTbs:=
;
param GB:=
;
param BB:=
;

**Comment by bukhsh** [comment]
Hi,
Many thanks for the question. I had a quick look at your data file and I noticed empty enteries for parameters (GB and BB). PYOMO do not link empty parameters (empty sets are ok). Removing these empty parameters will fix the error.

**Comment by maryammoh68** [comment]
Dear Bukhsh,
Thank you very much! I do not know why have I missed those parameters!
I will try it and write the answer again.
